### DISCUSSION TITLE
the best procedure to add/receive file into encrypted volume?

### DISCUSSION THREAD

**Discussion by braindevices** [hardcore-sushi/DroidFS#266] [Q&A]
This is a wonderful project. It is quite straightforward to move clear text file into the volume.
However, it seems not very easy to move gpg encrypted content into/out the volume.
I basically want to be able to use openkeychain to transfer some files without putting clear text on disk or clipboard.

The most straightforward way will be allow the open-keychain to read/save data on the volume. However, I cannot do this. The `open with external app` does not allow me to choose open-keychain.

Any suggestion in this situation? Or maybe this can be feature request?

**Comment by hardcore-sushi** [comment]
Why can't you simply select "Share" in OpenKeychain instead of "Save" and then choose "Encrypt with DroidFS"?
From within DroidFS, you can select a file and press "Share" to open an encrypted file with OpenKeychain (this requires enabling the "Allow sharing files via the android share menu" unsafe feature)

Exposed volumes might also be an option.

**Comment by braindevices** [comment]
Thanks a lot for the suggestion, the sharing trick of DroidFS->openkeychain works fine.

However, unfortunately openkeychain->DroidFS does not work.

1. in openkeychain click Encrypt/decrypt
2. choose a gpg file to decrypt
3. click share
4. choose encrypt with droidfs
5. choose/unlock a vault path
6. click on check mark on the top
7. Error: Failed to handle the share request.

Is this a bug? I guess DroidFS should provide a UI to add file name, otherwise the share will always fail since the content shared by openkeychain does not have filename at all.

Problem is here: https://github.com/hardcore-sushi/DroidFS/blob/967d4551c57abbc9febd5f23f6a7fd1af4118b89/app/src/main/java/sushi/hardcore/droidfs/explorers/ExplorerActivityDrop.kt#L38C43-L38C79

Instead of throw error, we have to handle the uri==null and try receivingIntent.getData() to get the actual content uri.

Show I create an issue for this?

**Comment by hardcore-sushi** [comment]
Importing decrypted **files** works fine for me. However, I got the same error as you when clicking on "Read from clipboard" in OpenKeychain. I'll try to fix that.

**Comment by hardcore-sushi** [comment]
This is now fixed with bd60e626350c9ff7b2ceb091af284df524891581
